bpfman/src/multiprog/tc.rs

```diff
@@ -165,19 +172,22 @@ impl TcDispatcher {
             ));
         }
 
-        let base = match direction {
-            Ingress => RTDIR_FS_TC_INGRESS,
-            Egress => RTDIR_FS_TC_EGRESS,
-        };
-        let path = format!("{base}/dispatcher_{if_index}_{revision}");
+        let path = tc_dispatcher_rev_path!(direction, netns.clone(), if_index, revision);
         fs::create_dir_all(path).unwrap();
 
         self.loader = Some(loader);
         self.set_num_extensions(extensions.len())?;
         self.set_program_name(TC_DISPATCHER_PROGRAM_NAME)?;
 
         self.attach_extensions(&mut extensions)?;
-        self.attach(root_db, old_dispatcher).await?;
+
+        if let Some(netns) = netns {
+            let _netns_guard = set_netns(netns)?;
```

Comment:
I'm assuming `drop()` is called automatically. Is it when the code leaves this if statement, the function, or other? May want to sprinkle some comments around this usage, because there is also the `set_netns()` function that writes the value to the DB. Someone skimming the code might miss that these are two different functions and this one is entering a network namespace.

Reply:
Yes, drop is called automatically when _netns_guard goes out of scope.  In this case, that's just inside the if statement.  I should probably change the function name so that there's no confusion.